### PR TITLE
DRAFT: Wrong data

### PR DESCRIPTION
**Pull request by TryHarder01**
draft of blog suggested in #104

### CODE REVIEW

File: content/post/wrong-data.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+I remember someone noting a report was wrong because the sales numbers didn’t make sense. How could we have sales for an item that hadn’t been released yet? Proof the report was wrong? Nope. The server logs showed an affiliate partner had published a direct link to an item we hadn’t made live yet. The report was correct, but by exploring the source of truth we were able to figure out something more meaningful for the company.
+
+## It really is wrong
+If you are giving your screen the “what the heck” stare, and find out that you goofed in your code, give yourself a pat on the back -- seriously. It’s easy enough to write the code and ship it, but to sanity check it and have the business context to see numbers don’t pass the sniff test takes practice. Catching “wrong” because you know what the numbers should be is an accomplishment. 
+
+You can up your chances at catching wrong data here by finding test groups and present your findings to them. 
+The data is confusing (but not wrong)
```
Comment (mikekaminsky) [inline content/post/wrong-data.md]: Is this supposed to be a header?

File: content/post/wrong-data.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+## Teach the customer; fix the product
+Finding the reason someone thinks a data point is wrong is key to applying the right solution, but It should not be used to offload responsibility. If data consumers look at a report and get the wrong idea of the business logic under the hood, that is as much of a problem as the math being wrong in a function. 
+
+Reporting tools are great for a business (see: [_Reporting is a Gateway Drug_](https://www.locallyoptimistic.com/post/reporting-is-a-gateway-drug/)), but with sharing information comes the reponsibility to help people make sense of it, and to have the right expectations. If you are digging into the data every day with code you’re all too familiar with the imperfections of the data stores you are touching. It’s helpful to keep clarifying what words mean, and what the data is tracking under the hood as you present it. For instance, if you talk about site sessions, share in writing what exactly a site session is. Sure everyone understands the concept of a session, but do they all have the same definition as the site tracking tool you’re using for your analysis?
```
Comment (mikekaminsky) [inline content/post/wrong-data.md]: I think it'd be better not to end with a question -- maybe re-work the paragraph so you lead with the question and end with the recommendation / answer.

Comment (mikekaminsky) [reply content/post/wrong-data.md]: This also might be a good spot to mention the [data quality flywheel](https://www.locallyoptimistic.com/post/data-dies-in-darkness/) (and give me that sweet sweet backlink)

File: content/post/wrong-data.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+publishDate: "????"
+cover: "????"
+draft: true
+tags: 
+ - data-integrity
+ - communication
+ - errors
+---
+
+Let us explore what it means for data to be wrong. For now, think of it from the user’s reaction only. It is a data point, trend line, or report that makes the user lean into their screen, squint, tilt their head a bit, and say, “nooooo, that’s wrong.”
```
Comment (mikekaminsky) [inline content/post/wrong-data.md]: Use of the open imperative in the first sentence feels odd to me.

> Let us explore what it means for data to be wrong.

could become

> What does it mean when data are "wrong"?

or

> Analytics stakeholders often accuse data of being "wrong" even when they are not.

Comment (mikekaminsky) [reply content/post/wrong-data.md]: Now that I think about it, probably better to re-work these lead-in paragraphs (what show up in the preview) to be a bit more thesis-driven (i.e., spelling out the conclusion of the post). Perhaps some of the language here would feel less odd if it were further down in the post.

File: content/post/wrong-data.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+draft: true
+tags: 
+ - data-integrity
+ - communication
+ - errors
+---
+
+Let us explore what it means for data to be wrong. For now, think of it from the user’s reaction only. It is a data point, trend line, or report that makes the user lean into their screen, squint, tilt their head a bit, and say, “nooooo, that’s wrong.”  
+
+Wanting to jump in and “fix” it is a natural feeling, especially for people who like the technical side of data. But let me make the case to pause for a beat and think through what “wrong” could mean before you head off fixing something that is not actually broken.
```
Comment (mikekaminsky) [inline content/post/wrong-data.md]: I think you can strengthen the second sentence

> But let me make the case to

 could become

> I recommend pausing...

or 

> You should pause...

File: content/post/wrong-data.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+ - data-integrity
+ - communication
+ - errors
+---
+
+Let us explore what it means for data to be wrong. For now, think of it from the user’s reaction only. It is a data point, trend line, or report that makes the user lean into their screen, squint, tilt their head a bit, and say, “nooooo, that’s wrong.”  
+
+Wanting to jump in and “fix” it is a natural feeling, especially for people who like the technical side of data. But let me make the case to pause for a beat and think through what “wrong” could mean before you head off fixing something that is not actually broken. 
+<!--more-->
+I propose there are three common situations in which data is considered wrong, and only one of them would fit a literal definition of “wrong.” They are (in order from least to most common):
```
Comment (mikekaminsky) [inline content/post/wrong-data.md]: Strike "I propose"

Comment (mikekaminsky) [reply content/post/wrong-data.md]: >There are three common situations in which data may be considered wrong, and the appropriate response to each situation is different.

File: content/post/wrong-data.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+<ul>
+<li> The data consumer is using a faulty source to “check” the data.
+<li> The data is literally wrong.
```
Comment (mikekaminsky) [inline content/post/wrong-data.md]: Propose striking "literally wrong" in favor of "there is a bug in the code" or "the source data are incorrect".

File: content/post/wrong-data.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+Wanting to jump in and “fix” it is a natural feeling, especially for people who like the technical side of data. But let me make the case to pause for a beat and think through what “wrong” could mean before you head off fixing something that is not actually broken. 
+<!--more-->
+I propose there are three common situations in which data is considered wrong, and only one of them would fit a literal definition of “wrong.” They are (in order from least to most common):
+
+<ul>
+<li> The data consumer is using a faulty source to “check” the data.
+<li> The data is literally wrong. 
+<li> The data consumer misunderstands what the data represents. 
+</ul>
```
Comment (mikekaminsky) [inline content/post/wrong-data.md]: Any reason you use the HTML here instead of using the markdown `*`?

File: content/post/wrong-data.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+## Begin with questions
+Recall the saying, “measure twice, cut once.” 
+
+Ask these two questions to the person who noticed the data being wrong (answer them yourself if you spotted the issue): What result would you expect to see here? Where would you check this number to know it’s right? A number missing a negative sign requires a solution miles away from a bug showing your sales a tenth of the actual number or off by a unit or two.
```
Comment (mikekaminsky) [inline content/post/wrong-data.md]: I'd propose you use bullets here for the visual call-out 

> * What result would you expect to see here? 
> * Where would you check this number to know it’s right?

Comment (mikekaminsky) [reply content/post/wrong-data.md]: > requires a solution miles away

could become

> requires a very different solution than

Comment (mikekaminsky) [reply content/post/wrong-data.md]: Actually this whole sentence was a little confusing to read and should be simplified:

>A number missing a negative sign requires a solution miles away from a bug showing your sales a tenth of the actual number or off by a unit or two.

Maybe

> Different types of problems require different solutions. You'll start different places when there's a negative sign missing versus when the number is off by an order of magnitude versus when the number is off by a few units.

File: content/post/wrong-data.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+Every time I look my old code I have new ideas for pretting it up, but when you’re on the hunt for the cause of wrong data keep an eye on the prize. Make time to deal with technical debt another time.
```
Comment (mikekaminsky) [inline content/post/wrong-data.md]: I'd consider combining this point with your lead paragraph. It feels a little odd that you start with "measure twice, cut once" but then end with "don't get distracted." I think this section would flow a little better if those ideas were married and kept consistent.

File: content/post/wrong-data.md [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+</ul>
+
+## Begin with questions
+Recall the saying, “measure twice, cut once.” 
+
+Ask these two questions to the person who noticed the data being wrong (answer them yourself if you spotted the issue): What result would you expect to see here? Where would you check this number to know it’s right? A number missing a negative sign requires a solution miles away from a bug showing your sales a tenth of the actual number or off by a unit or two.
+
+Every time I look my old code I have new ideas for pretting it up, but when you’re on the hunt for the cause of wrong data keep an eye on the prize. Make time to deal with technical debt another time. 
+
+## Other data is wrong
```
Comment (mikekaminsky) [inline content/post/wrong-data.md]: If it's the least common, maybe you should move it to the end?

File: content/post/wrong-data.md [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+This is the least common of the three types of wrong, but common enough that it’s worth checking. One place you might see this is in web site statistics. Suppose you have an internal tracking system tracking site visits, and also use an external tracking tool. On your way to the office an email hits you -- “the site session count is wrong.” (_note: emails arriving between 5am and when you walk into the office are almost never good. No one gets an email saying “Hey we decided to give you a raise.”_).
```
Comment (mikekaminsky) [inline content/post/wrong-data.md]: Strike the trailing parenthetical here or move it into a footnote.

File: content/post/wrong-data.md [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+
+## Other data is wrong
+This is the least common of the three types of wrong, but common enough that it’s worth checking. One place you might see this is in web site statistics. Suppose you have an internal tracking system tracking site visits, and also use an external tracking tool. On your way to the office an email hits you -- “the site session count is wrong.” (_note: emails arriving between 5am and when you walk into the office are almost never good. No one gets an email saying “Hey we decided to give you a raise.”_). 
+
+Someone shows you how the internal traffic number is 10x more than the external tracking tool. But can you trust that source? Maybe the service was down for an hour last night which caused it’s number to undercount your traffic. A quick check of the recent history might show you if there’s cause for alarm. The real problem might be the tracking of the secondary source.
+
+Maybe someone expected the numbers to match up exactly, which makes sense but also an unrealistic goal. If the two numbers are off by less than a percent it’s likely noise in how the two systems are tracking events.
```
Comment (mikekaminsky) [inline content/post/wrong-data.md]: but *is* also

Comment (mikekaminsky) [reply content/post/wrong-data.md]: Idea: could be useful to talk about a framework for thinking about what types of systems / when is it okay for a tolerance of 1% vs 10% vs 0%.

File: content/post/wrong-data.md [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+
+Maybe someone expected the numbers to match up exactly, which makes sense but also an unrealistic goal. If the two numbers are off by less than a percent it’s likely noise in how the two systems are tracking events. 
+
+I remember someone noting a report was wrong because the sales numbers didn’t make sense. How could we have sales for an item that hadn’t been released yet? Proof the report was wrong? Nope. The server logs showed an affiliate partner had published a direct link to an item we hadn’t made live yet. The report was correct, but by exploring the source of truth we were able to figure out something more meaningful for the company.
+
+## It really is wrong
+If you are giving your screen the “what the heck” stare, and find out that you goofed in your code, give yourself a pat on the back -- seriously. It’s easy enough to write the code and ship it, but to sanity check it and have the business context to see numbers don’t pass the sniff test takes practice. Catching “wrong” because you know what the numbers should be is an accomplishment.
```
Comment (mikekaminsky) [inline content/post/wrong-data.md]: I think it's also important to have a plan for communicating with stakeholders who may be affected. Being upfront and honest about mistakes is important for building a healthy analytics culture!

File: content/post/wrong-data.md (hunk 2 above)
Comment (mikekaminsky) [reply content/post/wrong-data.md]: I feel like we need a summary section at the end, maybe under a `## Conclusions` header that ties everything up nicely.

Comment (TryHarder01) [reply content/post/wrong-data.md]: done, not pushed yet.

File: content/post/wrong-data.md (hunk 3 above)
Comment (TryHarder01) [reply content/post/wrong-data.md]: done, not pushed yet.

File: content/post/wrong-data.md (hunk 4 above)
Comment (TryHarder01) [reply content/post/wrong-data.md]: done, not pushed yet.

File: content/post/wrong-data.md (hunk 11 above)
Comment (TryHarder01) [reply content/post/wrong-data.md]: done, not pushed.

File: content/post/wrong-data.md (hunk 12 above)
Comment (TryHarder01) [reply content/post/wrong-data.md]: done, not pushed.

File: content/post/wrong-data.md (hunk 10 above)
Comment (TryHarder01) [reply content/post/wrong-data.md]: done, not pushed

**Comment by TryHarder01** [comment]
@mikekaminsky addressed all your edits.

**Comment by TryHarder01** [comment]
Ughfff, @mikekaminsky ... I noticed that since I took the text into Google Docs to edit it (with a spell checker and with a friend), it killed the tracking on where your comments were tied to specific parts of the blog. 

Sorry about that.

**Pull request merged**